### PR TITLE
787: Adding probation rules and calculation contracted hours for the …

### PR DESCRIPTION
**Pull request by dconey646**
…workload owner. Both individual and team. Updated tests

### CODE REVIEW

File: app/services/get-overview.js [hunk 1]
```diff
@@ -16,18 +18,34 @@ module.exports = function (id, organisationLevel) {
...
-    }
+    result.overviewDetails = calculateCapacityPercentage(results)
     result.breadcrumbs = getBreadcrumbs(id, organisationLevel)
     result.title = result.breadcrumbs[0].title
     result.subTitle = organisationalUnitType.displayText
     return result
   })
 }
+
+var calculateCapacityPercentage = function (results) {
```
Comment (caoimheb) [inline app/services/get-overview.js]: Don't think `calculateCapacityPercentage` should be also calling `getContractedHours`. Possibly a rename? Or take contracted hours out of this function

File: app/services/data/get-individual-overview.js [hunk 2]
```diff
@@ -15,7 +16,9 @@ module.exports = function (id) {
            'workload_points_calculations.total_points AS totalPoints',
            'workload.total_cases AS cases',
            'workload_owner.contracted_hours AS contractedHours',
-           'workload_points_calculations.reduction_hours AS reduction')
+           'workload_points_calculations.reduction_hours AS reduction',
+           'workload_points.default_contracted_hours_po AS defaultContractedHoursPo',
+           'workload_points.default_contracted_hours_pso AS defaultContractedHoursPso')
```
Comment (caoimheb) [inline app/services/data/get-individual-overview.js]: Still not completely sure on best way to proceed with these bands. Currently they only have one default hours for all bands. Do we confine it to one? If we want more, do we need to include all? Just a thought...

File: test/unit/services/test-get-overview.js [hunk 3]
```diff
@@ -16,7 +16,10 @@ const OVERVIEW = {
   totalPoints: 40,
   cases: '2',
   hours: '3',
-  reduction: '4'
+  reduction: '4',
+  contractedHours: 0,
```
Comment (caoimheb) [inline test/unit/services/test-get-overview.js]: Should `contractedHours` returned in the results object from `getOverview` not have been replaced with the default value?

File: package.json [hunk 4]
```diff
@@ -85,6 +85,7 @@
     "serve-favicon": "2.4.x",
     "underscore": "^1.8.3",
     "util": "^0.10.3",
-    "validator": "^7.0.0"
+    "validator": "^7.0.0",
+    "wmt-probation-rules": "ministryofjustice/wmt-probation-rules.git#tech/wmt-787/make-default-contracted-hours-more-generic"
```
Comment (dconey646) [inline package.json]: Yea, so once probation rules is merged in, I will change to the newest version before this PR gets merged.

File: app/services/get-overview.js (hunk 1 above)
Comment (dconey646) [inline app/services/get-overview.js]: Good shout. I will rename

File: test/unit/services/test-get-overview.js (hunk 3 above)
Comment (dconey646) [inline test/unit/services/test-get-overview.js]: Yep, it makes more sense to have default hours.

File: app/services/get-overview.js [hunk 5]
```diff
@@ -16,18 +18,38 @@ module.exports = function (id, organisationLevel) {
...
-    }
+    result.overviewDetails = calcualteValues(results)
     result.breadcrumbs = getBreadcrumbs(id, organisationLevel)
     result.title = result.breadcrumbs[0].title
     result.subTitle = organisationalUnitType.displayText
     return result
   })
 }
+
+var calcualteValues = function (results) {
```
Comment (caoimheb) [inline app/services/get-overview.js]: Typo in function name

**Pull request merged**